### PR TITLE
Luacheck ci

### PR DESCRIPTION
**Pull request by BuckarooBanzay**
this PR adds:

* the remaining luacheck fixes
* a github action for luacheck execution on PR's and commits

### CODE REVIEW

File: .luacheckrc [hunk 1]
```diff
@@ -10,9 +9,21 @@ read_globals = {
...
 globals = {
-	"minetest"
+	"minetest",
+	-- mod namespace
+	"areas",
+	-- legacy
+	"IsPlayerNodeOwner",
+	"GetNodeOwnerName",
+	"HasOwner",
+	"owner_defs"
```
Comment (SmallJoker) [inline .luacheckrc]: How can these be acceptable globals? I'd prefer that luacheck reports those as coding errors.

Comment (BuckarooBanzay) [reply .luacheckrc]: legacy globals... well, i would like to remove them but i guess they are in use _somewhere_

https://github.com/minetest-mods/areas/blob/448fe3ebf18af48f8ef69738c2801c724d08bf10/legacy.lua#L110-L112

Comment (SmallJoker) [reply .luacheckrc]: These likely have their origins in the `node_ownership` mod and should be removed at least since 6 years.

**Comment by BuckarooBanzay** [comment]
legacy globals are removed, the `legacy.lua` file has still 2 (ignored) issues:

```
    legacy.lua:83:2: loop is executed at most once
    legacy.lua:104:2: loop is executed at most once
```

but IMO: they should be done in a later PR, goal of this was initially to introduce luacheck in a ci pipeline. But this grew to a legacy cleanup spree...

**Review by SmallJoker — APPROVED** [review]
Thank you.

**Comment by SmallJoker** [comment]
Will merge in the next few days unless there are objections by other devs.

File: .github/workflows/luacheck.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+on: [push, pull_request]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
```
Comment (Panquesito7) [inline .github/workflows/luacheck.yml]: I think it's best to use `v2` or even `master`.
This one is old and is slower than `v2`.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: .github/workflows/luacheck.yml (hunk 2 above)
Comment (BuckarooBanzay) [reply .github/workflows/luacheck.yml]: @Panquesito7 i've switched to `v2` for the actions/checkout image :+1: 
**Just fyi**: i'm thinking about creating a dedicated action for luachecks like this, the `apt-get` and `luarocks` are executed every time this gets invoked, not best in regards for resource-usage but it works for all my other projects :shrug: (and yes: i know the "factorio" luacheck action but i don't like the hardcoded absolute URL for the `.luacheckrc` file)

Comment (Panquesito7) [reply .github/workflows/luacheck.yml]: You can use Factorio LuaCheck Action without specifying a `.luacheckrc` URL:

              luacheckrc_url: ""

See Roang-zero1/factorio-mod-luacheck#4 for more information. 🙂

**Pull request merged**